### PR TITLE
Updated readme, removed commented code

### PR DESCRIPTION
**Pull request by jutonz**
New Overview, Installation, and Usage sections in readme.

Removed some commented code from index.js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,10 @@
-# service-maker
+# service-maker [![Build Status](https://travis-ci.org/inetCatapult/service-maker.svg?branch=master)](https://travis-ci.org/inetCatapult/service-maker)
+
+## Overview 
+Service Maker makes interacting with AWS easier. It does this by providing a set of sane defaults by which news AWS instances are created, so rather than specifying each individual setting required to spin up a new instance, all you have to do is tell Service Maker the important stuff. It takes care of the rest.
```
Comment (wtcross) [inline README.md]: Awesome!

File: index.js [hunk 2]
```diff
@@ -22,7 +22,6 @@ server.route({
 , path: '/new'
```
Comment (wtcross) [inline index.js]: What do you think about updating the routes in this PR?

`/v1/instances` for example

File: index.js [hunk 3]
```diff
@@ -22,7 +22,6 @@ server.route({
...
 , handler: function(request, reply) {
     var payload = JSON.parse(request.payload);
-    // reply(payload).code(201);
 
     if (payload.timeToLive === undefined || payload.size === undefined || payload.sshKeypair === undefined) {
```
Comment (wtcross) [inline index.js]: Using a [Joi](https://github.com/hapijs/joi) would be better here.

**Comment by wtcross** [comment]
Example Joi validator:

```javascript
// This is a Joi schema
var INSTANCE_SCHEMA = Joi.object().keys({
  timeToLive: Joi.number().integer().min(0), // duration in seconds
  size: Joi.string().valid("micro", "small", "medium", "large").default("micro"),
  sshKeyPair: Joi.string().required()
});

// Use the schema to validate a payload
var validated = Joi.validate(request.payload, INSTANCE_SCHEMA);
if (validated.error) {
  reply(Boom.badRequest(validated.error.message));
} else {
  var instance = validated.value;
  // do cool stuff with instance
}
```

**Comment by jutonz** [comment]
Awesome, thanks Tyler! I'll incorporate all of your comments and make another PR.

**Comment by wtcross** [comment]
All you have to do is push changes to your working branch. It will automagically update the PR :smile:

**Comment by wtcross** [comment]
That is, push to `jutonz:master`

**Comment by jutonz** [comment]
Oh, I didn't know that. Haha I'll do that then--thanks again!

**Comment by wtcross** [comment]
PRs are great tools for having and capturing conversation about new code, seeing the changes in the same PR makes it easier to follow the history.

**Comment by jutonz** [comment]
Yeah, that's definitely something I'm learning! I didn't realize how flexible they are. Thanks again for your feedback!

**Pull request merged**